pkg/model/awsmodel/autoscalinggroup.go

```diff
@@ -117,6 +117,11 @@ func (b *AutoscalingGroupModelBuilder) buildLaunchTemplateTask(c *fi.ModelBuilde
 		return nil, fmt.Errorf("error building cloud tags: %v", err)
 	}
 
+	cpuCredits := fi.StringValue(ig.Spec.CPUCredits)
+	if cpuCredits == "" {
+		cpuCredits = "unlimited"
+	}
+
```

Comment:
I think this should be moved at the end of the function and keep similar style.

Reply:
done, though some of the later assignments in that function are unnecessary. We're checking if a pointer is non-nil and conditionally assigning it to the task's pointer field. We could unconditionally assign it (where my CPUCredits assignment originally was) without any change in behavior. Maybe a cleanup opportunity after the launch configuration removal is merged.

Reply:
👍